pycket/pycket-lang/zoTransform.rkt

```diff
@@ -0,0 +1,581 @@
+               (cons l (with-handlers ([exn:fail? (lambda (x) #f)])
+                         (compile l))))))]
+        [table (make-hash)])
+    (for ([b (in-list bindings)])
+      (let ([v (and (cdr b)
+                    (zo-parse 
+                     (open-input-bytes
+                      (with-output-to-bytes
+                          (λ () (write (cdr b)))))))])
+        (let ([n (match v
+                   [(struct compilation-top (_ _ prefix (struct primval (n)))) n]
+                   [else #f])])
+          (hash-set! table n (car b)))))
+    table))
+
+(define (compile-json config language topmod body1 top-require-forms body-forms pycket?)
+  (let ([wholeBody (append top-require-forms body-forms)])
+    (hash* 'language (list language)
+           'module-name topmod
+           'config config
+           'body-forms (if pycket?
+                           wholeBody
+                           (cons body1 wholeBody)))))
+  
+(define (handle-def-values def-values-form toplevels  localref-stack)
+  (let ((ids (def-values-ids def-values-form))
+        (rhs (def-values-rhs def-values-form)))
+    (cond
+      ((or (not (= 1 (length ids))) (not (toplevel? (car ids))))
+       (error 'handle-def-values "look into multiple toplevels"))
+      (else
+       (let* ((toplevel-form (car ids))
+              (pos (toplevel-pos toplevel-form))
+              (sym (list-ref toplevels pos))
+              (symstr (if (symbol? sym) (symbol->string sym) sym))
+              )
+         (hash* 'define-values (list symstr)
+                'define-values-names (list symstr)
+                'define-values-body (to-ast-single rhs toplevels
+                                                   (cons symstr localref-stack))))))))
+
+(define (handle-if if-form toplevels localref-stack)
+  (let ((test (branch-test if-form))
+        (then (branch-then if-form))
+        (else (branch-else if-form)))
+    (hash*
+     'test (to-ast-single test toplevels localref-stack)
+     'then (to-ast-single then toplevels localref-stack)
+     'else (to-ast-single else toplevels localref-stack))))
+
+;; TODO: we'll probably need a more precise one
+(define (handle-number racket-num)
+  (cond
+    [(integer? racket-num)
+     (hash* 'integer (number->string racket-num))]
+    [(real? racket-num)
+     (if (inexact? racket-num)
+         (hash* 'real racket-num)
+         (let ;; assumes it's an exact rational
+             ([num (numerator racket-num)]
+              [den (denominator racket-num)])
+           (hash* 'numerator (handle-number num)
+                  'denominator (handle-number den))))
+         ]
+    [else ; this part assumes (for the moment) it's a complex num
+     (let ([real (real-part racket-num)]
+           [imag (imag-part racket-num)])
+       (hash* 'real-part (handle-number real)
+              'imag-part (handle-number imag)))]))
+
+(define (handle-boolean racket-bool)
+  (if racket-bool
```

Comment:
I remember at some point pycket complained about `(hash* 'quote racket-bool)`, but apparently I have confused it for another error, it works now.

The `(hash* x #f)` is rather strange since `(apply hash-set* (hash) '(x #f))` works, which is how hash* is implemented in expand.rkt